src/modules/sensors/vehicle_imu/VehicleIMU.cpp

```diff
@@ -638,4 +656,108 @@ void VehicleIMU::PrintStatus()
+	// Larger values cause a larger fraction of the learned biases to be used.
+	static constexpr float max_var_allowed = 1e-3f;
+	static constexpr float max_var_ratio = 1e2f;
+
+	if (_armed) {
+		static constexpr uint8_t accel_cal_size = sizeof(_accel_cal) / sizeof(_accel_cal[0]);
+
+		for (int i = 0; i < math::min(_estimator_sensor_bias_subs.size(), accel_cal_size); i++) {
+			estimator_sensor_bias_s estimator_sensor_bias;
+
+			if (_estimator_sensor_bias_subs[i].update(&estimator_sensor_bias)) {
+				// find corresponding accel calibration
+				if (_accel_calibration.device_id() == estimator_sensor_bias.accel_device_id) {
+					const Vector3f bias{estimator_sensor_bias.accel_bias};
+					const Vector3f bias_variance{estimator_sensor_bias.accel_bias_variance};
+
+					const bool valid = (hrt_elapsed_time(&estimator_sensor_bias.timestamp) < 1_s) &&
+							   (estimator_sensor_bias.accel_device_id != 0) &&
+							   estimator_sensor_bias.accel_bias_valid &&
+							   (bias_variance.max() < max_var_allowed) &&
+							   (bias_variance.max() < max_var_ratio * bias_variance.min());
+
+					if (valid) {
+						_accel_cal[i].device_id = estimator_sensor_bias.accel_device_id;
+						_accel_cal[i].accel_offset = bias;
+						_accel_cal[i].accel_bias_variance = bias_variance;
+						_accel_cal[i].valid = true;
+						_accel_cal_available = true;
+						break;
+
+					} else {
+						_accel_cal[i].valid = false;
+					}
+				}
+			}
+		}
+
+	} else if (_accel_cal_available) {
+		// not armed and accel cal available
+		bool calibration_param_save_needed = false;
+
+		bool initialised = false;
+		Vector3f bias_variance {};
+		Vector3f bias_estimate {};
+
+		// apply all valid saved offsets
+		for (int i = 0; i < ORB_MULTI_MAX_INSTANCES; i++) {
+			if ((_accel_calibration.device_id() != 0)
+			    && (_accel_cal[i].device_id == _accel_calibration.device_id())
+			    && _accel_cal[i].valid) {
+
+				if (!initialised) {
+					bias_variance = _accel_cal[i].accel_bias_variance;
+					bias_estimate = _accel_cal[i].accel_offset;
+					initialised = true;
+
+				} else {
+					for (int axis_index = 0; axis_index < 3; axis_index++) {
+						const float sum_of_variances = _accel_cal[i].accel_bias_variance(axis_index) + bias_variance(axis_index);
+						const float k1 = bias_variance(axis_index) / sum_of_variances;
+						const float k2 = _accel_cal[i].accel_bias_variance(axis_index) / sum_of_variances;
+						bias_estimate(axis_index) = k2 * bias_estimate(axis_index) + k1 * _accel_cal[i].accel_offset(axis_index);
+						bias_variance(axis_index) *= k2;
+					}
+				}
+			}
+		}
+
+		if (initialised && bias_estimate.longerThan(0.05f)) {
+			const Vector3f accel_cal_orig{_accel_calibration.offset()};
+			Vector3f accel_cal_offset{_accel_calibration.offset()};
+
+			for (int axis_index = 0; axis_index < 3; axis_index++) {
+				accel_cal_offset(axis_index) += bias_estimate(axis_index);
+			}
+
+			if (_accel_calibration.set_offset(accel_cal_offset)) {
```

Comment:
The offset is in the sensor frame. For mag calibration I added a helper in the calibration library.

https://github.com/PX4/PX4-Autopilot/blob/03bd5e0f4841debf5e45d0eadbfb28f131eaaeb1/src/lib/sensor_calibration/Magnetometer.hpp#L89-L93

Reply:
Fixed